src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -73,31 +82,154 @@ internal static TupleTypeSymbol ConstructTupleTypeSymbol(TupleTypeSymbol origina
 
             var numElements = elementTypes.Length;
 
-            // PROTOTYPE handling for greater than 8
-            if (numElements <= 1 || numElements > 7)
+            if (numElements <= 1)
             {
                 throw ExceptionUtilities.Unreachable;
             }
+            var underlyingTypeChain = GetTupleUnderlyingTypeChain(elementTypes, syntax, binder, diagnostics);
+
+            var fieldsBuilder = ArrayBuilder<TupleFieldSymbol>.GetInstance(numElements);
+
+            var hasChunks = numElements > 7;
+            if (!hasChunks)
+            {
+                // short tuple: each member gets a simple field symbol
+                for (int elementIndex = 0; elementIndex < numElements; elementIndex++)
+                {
+                    var wellKnownTypeMember = GetTupleTypeMember(numElements, elementIndex);
+                    var underlyingField = (FieldSymbol)Binder.GetWellKnownTypeMember(binder.Compilation, wellKnownTypeMember, diagnostics, syntax: syntax);
+
+                    var field = new TupleFieldSymbol(elementNames.IsDefault ? TupleMemberName(elementIndex) : elementNames[elementIndex],
+                                               this,
+                                               elementTypes[elementIndex],
+                                               OneOrMany.Create(underlyingField?.AsMember(underlyingTypeChain[0])));
+                    fieldsBuilder.Add(field);
+                }
+            }
+            else
+            {
+                var maxNumLinks = NumberOfValueTuples(numElements);
+
+                // long tuple: each member gets a chain of field symbols
+                for (int elementIndex = 0; elementIndex < numElements; elementIndex++)
+                {
+                    var numLinks = NumberOfValueTuples(elementIndex + 1);
+                    var chainBuilder = ArrayBuilder<FieldSymbol>.GetInstance(numLinks);
+
+                    for (var linkIndex = 0; linkIndex < numLinks; linkIndex++)
+                    {
+                        var linkArity = (linkIndex != maxNumLinks - 1) ? 8 : SizeOfLastValueTuples(numElements);
+                        var linkPosition = (linkIndex != numLinks - 1) ? 7 : elementIndex % 7;
```

Comment:
I didn't address this yet. I think fewer constant `7` appear after I address the first round of PR feedback. Let me know if you still think it is worthwhile.